components/layout/construct.rs

```diff
@@ -629,8 +630,12 @@ impl<'a> FlowConstructor<'a> {
     /// to happen.
     fn build_flow_for_nonfloated_block(&mut self, node: &ThreadSafeLayoutNode)
                                        -> ConstructionResult {
```

Comment:
Yes they can, I meant to start with the easy case first and follow up later.

I don’t know if this belongs in that `match` since multicol apply to anything that would otherwise be a block container, so you first have to determine that.

On that topic, http://dev.w3.org/csswg/css-display/ splits up the `display` property into orthogonal `display-outside` and `display-inside`. Regardless of supporting these properties, I think we should have that split internally: what a box is externally (inline-level, block-level, floating, abspos, flex item, table cell, …) should be mostly independent of what it’s inside (block container, multicol, flex, table, …).